cardano-cli/src/Cardano/CLI/Types/Errors/DebugCmdError.hs

```diff
@@ -7,13 +7,46 @@ where
 
 import           Cardano.Api
 
+import           Cardano.Chain.Genesis
 import           Cardano.CLI.Types.Errors.TxCmdError
 
+import           Data.Text (Text)
+import qualified Data.Text.Lazy.Builder as Text
+import           Formatting.Buildable (build)
+
 data DebugCmdError
   = DebugCmdFailed
+  | -- | @DebugNodeConfigGenesisDataCmdError filepath error@ represents an error when
+    -- reading the node configuration at @filepath@
+    DebugNodeConfigGenesisDataCmdError !FilePath !GenesisDataError
+  | -- | @DebugNodeConfigReadCmdError filepath error@ represents an error when
+    -- reading the node configuration at @filepath@
+    DebugNodeConfigReadCmdError !FilePath !String
+  | -- | @DebugNodeConfigWrongGenesisHashCmdError filepath era actualHash expectedHash@ represents a user error
+    -- that the genesis hash for @era@ in @filepath@ is @actualHash@, whereas it should be @expectedHash@
+    DebugNodeConfigWrongGenesisHashCmdError !FilePath AnyCardanoEra !Text !Text
```

Comment:
Some haddocks to describe what `Text` values are would be useful.

Reply:
Added haddocks to the individual arguments :+1: